### ISSUE TITLE
Current master not working with Ecto/Phoenix

### ISSUE THREAD

**Issue by dantswain** [elixir-ecto/postgrex#270]
I tried pulling master (7d008e1283b606254495497326c3b060f9e77ecb at time of writing) into my project because I needed the geometry types.   Running `mix test` I get the following output.  I get similar with `iex -S mix`.  I'll investigate.  I noticed a couple compile warnings as well.  This was all working as of d5c627d486c8a20728e56443424dc85b1a260ce5 

```
19:28:21.670 [error] GenServer #PID<0.349.0> terminating
** (UndefinedFunctionError) function true.find/2 is undefined (module true is not available)
    true.find(%Postgrex.TypeInfo{array_elem: 0, base_type: 0, comp_elems: [], input: "boolin", oid: 16, output: "boolout", receive: "boolrecv", send: "boolsend", type: "bool"}, :any)
    (postgrex) lib/postgrex/types.ex:111: Postgrex.Types.find/4
    (postgrex) lib/postgrex/types.ex:104: anonymous fn/4 in Postgrex.Types.associate_type_infos/2
    (elixir) lib/enum.ex:1623: Enum."-reduce/3-lists^foldl/2-0-"/3
    (postgrex) lib/postgrex/types.ex:103: Postgrex.Types.associate_type_infos/2
    (postgrex) lib/postgrex/type_server.ex:122: Postgrex.TypeServer.associate/2
    (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
    (stdlib) gen_server.erl:647: :gen_server.handle_msg/5

19:28:21.681 [error] GenServer #PID<0.348.0> terminating
** (stop) exited in: :gen_server.call(#PID<0.346.0>, {:checkout, #Reference<0.0.3.13972>, true, :infinity}, 5000)
    ** (EXIT) an exception was raised:
        ** (UndefinedFunctionError) function true.find/2 is undefined (module true is not available)
            true.find(%Postgrex.TypeInfo{array_elem: 0, base_type: 0, comp_elems: [], input: "boolin", oid: 16, output: "boolout", receive: "boolrecv", send: "boolsend", type: "bool"}, :any)
            (postgrex) lib/postgrex/types.ex:111: Postgrex.Types.find/4
            (postgrex) lib/postgrex/types.ex:104: anonymous fn/4 in Postgrex.Types.associate_type_infos/2
            (elixir) lib/enum.ex:1623: Enum."-reduce/3-lists^foldl/2-0-"/3
            (postgrex) lib/postgrex/types.ex:103: Postgrex.Types.associate_type_infos/2
            (postgrex) lib/postgrex/type_server.ex:122: Postgrex.TypeServer.associate/2
            (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
            (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
    (db_connection) lib/db_connection/poolboy.ex:112: DBConnection.Poolboy.checkout/3
    (db_connection) lib/db_connection/ownership/proxy.ex:108: DBConnection.Ownership.Proxy.handle_call/3
    (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
    (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
** (DBConnection.ConnectionError) failed to checkout using DBConnection.Poolboy
    (db_connection) lib/db_connection.ex:925: DBConnection.checkout/2
    (db_connection) lib/db_connection.ex:741: DBConnection.run/3
    (db_connection) lib/db_connection.ex:1132: DBConnection.run_meter/3
    (db_connection) lib/db_connection.ex:584: DBConnection.prepare_execute/4
    (ecto) lib/ecto/adapters/postgres/connection.ex:91: Ecto.Adapters.Postgres.Connection.execute/4
    (ecto) lib/ecto/adapters/sql.ex:235: Ecto.Adapters.SQL.sql_call/6
    (ecto) lib/ecto/adapters/sql.ex:185: Ecto.Adapters.SQL.query!/5
    (ecto) lib/ecto/adapters/postgres.ex:71: Ecto.Adapters.Postgres.execute_ddl/3


19:28:21.768 [error] GenServer #PID<0.351.0> terminating
** (UndefinedFunctionError) function true.find/2 is undefined (module true is not available)
    true.find(%Postgrex.TypeInfo{array_elem: 0, base_type: 0, comp_elems: [], input: "boolin", oid: 16, output: "boolout", receive: "boolrecv", send: "boolsend", type: "bool"}, :any)
    (postgrex) lib/postgrex/types.ex:111: Postgrex.Types.find/4
    (postgrex) lib/postgrex/types.ex:104: anonymous fn/4 in Postgrex.Types.associate_type_infos/2
    (elixir) lib/enum.ex:1623: Enum."-reduce/3-lists^foldl/2-0-"/3
    (postgrex) lib/postgrex/types.ex:103: Postgrex.Types.associate_type_infos/2
    (postgrex) lib/postgrex/type_server.ex:122: Postgrex.TypeServer.associate/2
    (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
    (stdlib) gen_server.erl:647: :gen_server.handle_msg/5

19:28:21.771 [error] GenServer #PID<0.352.0> terminating
** (stop) exited in: :gen_server.call(#PID<0.350.0>, {:checkout, #Reference<0.0.1.12036>, true, :infinity}, 5000)
    ** (EXIT) an exception was raised:
        ** (UndefinedFunctionError) function true.find/2 is undefined (module true is not available)
            true.find(%Postgrex.TypeInfo{array_elem: 0, base_type: 0, comp_elems: [], input: "boolin", oid: 16, output: "boolout", receive: "boolrecv", send: "boolsend", type: "bool"}, :any)
            (postgrex) lib/postgrex/types.ex:111: Postgrex.Types.find/4
            (postgrex) lib/postgrex/types.ex:104: anonymous fn/4 in Postgrex.Types.associate_type_infos/2
            (elixir) lib/enum.ex:1623: Enum."-reduce/3-lists^foldl/2-0-"/3
            (postgrex) lib/postgrex/types.ex:103: Postgrex.Types.associate_type_infos/2
            (postgrex) lib/postgrex/type_server.ex:122: Postgrex.TypeServer.associate/2
            (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
            (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
    (db_connection) lib/db_connection/poolboy.ex:112: DBConnection.Poolboy.checkout/3
    (db_connection) lib/db_connection/ownership/proxy.ex:108: DBConnection.Ownership.Proxy.handle_call/3
    (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
    (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

**Comment by dantswain** [comment]
Closing as this appears to be in Ecto, not postgrex.

**Comment by josevalim** [comment]
@dantswain if you use Postgrex master and Ecto master, you should be fine to go. The CHANGELOG for both projects contains information about this and what is happening.

**Comment by dantswain** [comment]
Thanks, @josevalim that's what I ended up doing.

One thing that did happen that I don't see mentioned in the CHANGELOG on either project is `mix test` broke because it was trying to read `schema_migrations.version` as an integer and it's a string in my db (it's a Rails-owned db).  Since it's an existing db I just disabled that step in the `test` alias and was fine.  If that's not a known problem I could create an issue in Ecto but I assumed it was known?
